repository/src/main/java/com/epam/esm/repository/impl/TagRepositoryImpl.java

```diff
@@ -1,93 +1,82 @@
 package com.epam.esm.repository.impl;
 
 import com.epam.esm.entity.Tag;
-import com.epam.esm.exception.TagNotFoundException;
-import com.epam.esm.exception.TagOperationException;
+import com.epam.esm.exception.NotFoundException;
+import com.epam.esm.exception.ValidationException;
 import com.epam.esm.repository.TagRepository;
-import lombok.extern.slf4j.Slf4j;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.dao.DataAccessException;
-import org.springframework.dao.EmptyResultDataAccessException;
-import org.springframework.jdbc.core.BeanPropertyRowMapper;
-import org.springframework.jdbc.core.JdbcTemplate;
-import org.springframework.jdbc.support.GeneratedKeyHolder;
-import org.springframework.jdbc.support.KeyHolder;
+import com.epam.esm.validator.EntityValidator;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.persistence.TypedQuery;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Root;
 import org.springframework.stereotype.Repository;
 
-import java.sql.PreparedStatement;
 import java.util.List;
-import java.util.Objects;
 import java.util.Optional;
 
-@Slf4j
 @Repository
 public class TagRepositoryImpl implements TagRepository {
```

Comment:
I am not seeing method or query here to implement this functionality
<img width="553" alt="image" src="https://github.com/AkobirToshtemirov/GiftCertificates/assets/80680360/723b464d-bdd1-46d4-8ad6-c52ae900060e">